lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/util/ClosablesTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.plugin.base.util.Closables.closeAllSuppress;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertTrue;
+
+public class ClosablesTest
+{
+    @Test
+    public void testCloseAllSuppressNonThrowing()
+    {
+        RuntimeException rootException = new RuntimeException("root");
+        TestClosable nonThrowingClosable = new TestClosable(Optional.empty());
+        closeAllSuppress(rootException, nonThrowingClosable);
+        assertTrue(nonThrowingClosable.isClosed());
+        assertThat(rootException.getSuppressed()).isEmpty();
+    }
+
+    @Test
+    public void testCloseAllSuppressThrowingOther()
+    {
+        RuntimeException rootException = new RuntimeException("root");
+        RuntimeException closeException = new RuntimeException("close");
+        TestClosable nonThrowingClosable = new TestClosable(Optional.of(closeException));
+        closeAllSuppress(rootException, nonThrowingClosable);
+        assertTrue(nonThrowingClosable.isClosed());
+        assertThat(rootException.getSuppressed()).containsExactly(closeException);
+    }
+
+    @Test
+    public void testCloseAllSuppressThrowingRoot()
+    {
+        RuntimeException rootException = new RuntimeException("root");
+        TestClosable nonThrowingClosable = new TestClosable(Optional.of(rootException));
+        closeAllSuppress(rootException, nonThrowingClosable);
+        assertTrue(nonThrowingClosable.isClosed());
+        assertThat(rootException.getSuppressed()).isEmpty();
+    }
+
+    @Test
+    public void testCloseAllSuppressNullClosable()
+    {
+        RuntimeException rootException = new RuntimeException("root");
+        closeAllSuppress(rootException, (AutoCloseable) null);
```

Comment:
@findepi If I passed just `null` here (without the cast) it was internally interpreted as whole `closeables` array and `requireNonNull(closeables, "closeables is null")` check trigger. 
I do not think this is a big deal though. Why would someone use untyped null literal here?

Reply:
> I do not think this is a big deal though. Why would someone use untyped null literal here?

agreed, that would be wrong